aws-rds-cfn-common/src/main/java/software/amazon/rds/common/client/RdsUserAgentProvider.java

```diff
@@ -0,0 +1,47 @@
+package software.amazon.rds.common.client;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+import com.amazonaws.util.StringUtils;
+import software.amazon.rds.common.test.ExcludeFromJacocoGeneratedReport;
+
+public final class RdsUserAgentProvider {
+
+    public static final String SDK_CLIENT_USER_AGENT_PREFIX = "AWS/CloudFormation/RDS";
+
+    static final String VERSION_PROPERTIES_FILE_PATH = "version.properties";
+    static final String GIT_COMMIT_ID_ABBREV_PROPERTY = "git.commit.id.abbrev";
```

Comment:
Is it possible to add first 50/60 characters of git commit message? I think it might boost readability of UserAgent signature to operator that he do not need to return to git history?

Reply:
Given a commit message is a free-format UTF8 text, I won't really trust it: we can introduce more harm than help. any SCM is able to identify a commit by the first 4-6 characters of its SHA hash, given it has a high hash entropy. E.g. here Github finds my commit by the first 5 characters: https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-rds/commit/11210.